### PR TITLE
Patch crossbeam-epoch to avoid overhead

### PR DESCRIPTION
**Pull request by ryoqun**
#### Problem

fixes https://github.com/solana-labs/solana/issues/22603

our rayon usage might be peculiar enough to warrant patching...:

https://github.com/crossbeam-rs/crossbeam/issues/852#issuecomment-1180199333

#### Summary of Changes

Fixes #

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -91,5 +91,6 @@ exclude = [
     "programs/bpf",
 ]
 
-# This prevents a Travis CI error when building for Windows.
-resolver = "2"
```
[the remark this answers is not in the document]
Comment (ryoqun) [reply Cargo.toml:95]: fyi, we won't ever need to patch for windows builds. this patch is only relevant `solana-validator` and i don't think anyone ever want to run it on windows. :)

File: Cargo.lock [hunk 2]
```diff
@@ -968,8 +968,7 @@ dependencies = [
 [[package]]
 name = "crossbeam-epoch"
 version = "0.9.5"
```
Comment (ryoqun) [inline Cargo.lock:970]:
```
$ cat /home/ryoqun/.cargo/registry/src/github.com-1ecc6299db9ec823/crossbeam-epoch-0.9.5/.cargo_vcs_info.json 
{
  "git": {
    "sha1": "6d4cdd4daf9a897deef6cde9569f2fbf12c29bc5"
  }
}

```

File: Cargo.toml [hunk 3]
```diff
@@ -91,5 +91,6 @@ exclude = [
...
+# for details, see https://github.com/ryoqun/crossbeam/commit/fd279d707025f0e60951e429bf778b4813d1b6bf
```
Comment (ryoqun) [inline Cargo.toml:94]: for convenience: https://github.com/ryoqun/crossbeam/commit/fd279d707025f0e60951e429bf778b4813d1b6bf

File: Cargo.lock (hunk 2 above)
Comment (ryoqun) [reply Cargo.lock:970]: you can verify that there is barely minimum modifications by comparing the published commit rev with the parent commit rev of my patch commit: https://github.com/solana-labs/solana/pull/26555/files#r918444566

File: Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (willhickey) [reply Cargo.toml:95]: Yes, we can remove resolver=2 from Cargo.toml and add it back in during the windows build. Let me know if you'd like to proceed with this change and I'll open a PR to update the build action.

Here's the commit:
https://github.com/solana-labs/solana/commit/0b9eb201aa0490d98543f759e66bda0e69ceaa38

[the remark this answers is not in the document]
Comment (brennanwatt) [reply Cargo.toml:95]: Hey @willhickey - I'm going to help @ryoqun get this one across the finish line unless there are objections. If you can open the PR to update the build action, that would be great.

In the meantime, I'm going to kickoff a validator on MNB to run with this change

[the remark this answers is not in the document]
Comment (willhickey) [reply Cargo.toml:95]: Just created PR #26706

[the remark this answers is not in the document]
Comment (willhickey) [reply Cargo.toml:95]: PR #26706 is merged. This PR may need to be rebased. Let me know if there's anything else I can do to help here

**Comment by brennanwatt** [comment]
I've run a validator on MNB for multiple days w/o any issues (w/ this change on top of v1.10.32)

**Review by brennanwatt — APPROVED** [review]

**Comment by ryoqun** [comment]
a few hygiene are done:

- forked crossbeam to solana-labs and pushed my patch there (maybe cosmetic)
- squashed on the current tip of master instead of merge commit. (my preference).

**Review by brennanwatt — APPROVED** [review]

**Pull request merged**